<div style="background: #000;
            color: #FFF;
            margin: 0px;
                padding: 10px 0px 20px 0px;
            text-align: center; 
                ">
    <h1>Week 3 - Homework</h1>
</div>

<hr>

Exercise 1 due Monday.  

Exercise 2 due Tueday.

Exercise 3 due Wednesday.
<hr>

# Exercise 1

Write a for loop that sums the values of the following list and appends the different values to the correct list.

Remember what we learned about the value of an `int` and `str`.

Hint: try one part before the other.

In [1]:
# list to sum
list_vals = [82,True,'d',False,'v',7301,False,1,True,'x','F','w','Q','f',True]
#import hashlib;print("list_vals is correct" if hashlib.md5(str(list_vals).encode()).hexdigest() == '867f5d64395c44f016668cc442fb650c' else "list_vals is incorrect")

In [14]:
list_for_bools = []
list_for_ints = []
list_for_chars = []
sum_of_all = 0

for val in list_vals:
  if type(val) == int:
    list_for_ints.append(val)
    sum_of_all += val
  elif type(val) == bool:
    list_for_bools.append(val)
    sum_of_all += int(val)
  else:
    list_for_chars.append(val)
    sum_of_all += ord(val)
  


# use the following to output your results:
print(sum_of_all)
print(list_for_bools)
print(list_for_ints)
print(list_for_chars)

8097
[True, False, False, True, True]
[82, 7301, 1]
['d', 'v', 'x', 'F', 'w', 'Q', 'f']


### Final Answer: 
`sum == 8097`  
`len(list_for_bools) == 5`  
`len(list_for_ints) == 3`   
`len(list_for_ints) == 7`   

# Exercise 2

Given a tuple of strings. Find the following value `x` and print it. The loop should end when `x` is found

In [28]:
x = '------'

tuple_of_strings = ('steel', 'collect', 'capable', '------', 'shelter', 'language', 'mug', 'flowers')

# enter your code here:

found = False
print("Searching...")
while found == False:
  for item in tuple_of_strings:
    print("Current item:", item)
    if item == x:
      print("Item found!")
      found = True
      break

Searching...
Current item: steel
Current item: collect
Current item: capable
Current item: ------
Item found!


# Exercise 3

Create a game in python. 

Rules:  
    1. You have 3 tries to guess a secret number.   
    2. Each try you are prompted to input a number.  
    3. You get a success message if you guess correctly and stop looping, else you subtract a try and go again.
    

In [60]:
import random as random
##### use secret_number in your code but don't cheat
secret_number = random.randint(0,3)
user_guess = int(input("Guess a number 0 - 3: "))
num_guesses = 2

while num_guesses > 0 and user_guess != secret_number:
  print("Try again.", num_guesses, "guesses remaining.")
  user_guess = int(input("Guess another number 0 - 3: "))
  num_guesses -= 1
  
if user_guess == secret_number:
  print("You guessed the correct number!")
else:
  print("Better luck next time.")    

Guess a number 0 - 3: 3
Try again. 2 guesses remaining.
Guess another number 0 - 3: 3
Try again. 1 guesses remaining.
Guess another number 0 - 3: 3
Better luck next time.


# Bonus Exercise
Research List & Dict Comprehension and try these problems. Also leave a written explanation of both.

a. 
Create a list comprehension that prints the following.
```python
[0, '*',2, '*', 4, '*', 6]

```

Hint: start from the most basic list comprehension  

In [63]:
List = [item for item in [0, '*', 2, '*', 4, '*', 6]]  # Copies each item in the iterable target list
print(List)

[0, '*', 2, '*', 4, '*', 6]


b. Create a dict comprehension that prints the following:
```python
{'a': ['*', '*'], 'b': ['*', '*'],}

```


In [68]:
Dictionary = {items: ['*', '*'] for items in ('a', 'b')} # Assigns the list ['*', '*'] to each item in the tuple ('a', 'b')
print(Dictionary)

{'a': ['*', '*'], 'b': ['*', '*']}
